ledger/participant-integration-api/src/main/scala/platform/index/LedgerEndPoller.scala

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.index
+
+import akka.actor.Cancellable
+import akka.stream.scaladsl.{Keep, RestartSource, Sink, Source}
+import akka.stream.{KillSwitches, Materializer, RestartSettings, UniqueKillSwitch}
+import akka.{Done, NotUsed}
+import com.daml.logging.LoggingContext
+import com.daml.platform.store.appendonlydao.LedgerReadDao
+import com.daml.platform.store.backend.ParameterStorageBackend.LedgerEnd
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+
+/** Periodically polls the ledger end from the [[LedgerReadDao]]
+  * and updates the caches backing the Ledger API.
+  */
+private[index] class LedgerEndPoller(
+    ledgerReadDao: LedgerReadDao,
+    consume: LedgerEnd => Future[Unit],
+)(implicit mat: Materializer, loggingContext: LoggingContext) {
+
+  private val restartSettings =
+    RestartSettings(minBackoff = 1.second, maxBackoff = 10.seconds, randomFactor = 0.2)
```

Comment:
What's the strategy of having configuration constants in the codebase? Shouldn't it be configurable?

Reply:
I am not sure. Generally yes, I think exposing configuration parameters makes sense when the parameter
* is impacting the happy flow
* is configuring performance critical flows
* is toggling features in the participant
and other reasons as important that I might have missed.

On the other hand, I think we can omit bloating the CLI surface with configs for settings that are about the low-level details of the participant which are also not in the happy flow (like in this case). e.g. I don't see any specific reason on why we would ever want to tune these parameters based on the deployment context. Does it make sense?

Reply:
Yes it makes sense, agree. At least I was not expecting them to be passed across the whole configuration layer including CLI. It could be just a config object which is always provided as default argument to the constructor, then you don't need to touch this class whenever you want to use this class with different parametrisation (including tests).